File: Q01.java
import java.util.Scanner;

public class Q01 {


    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Quantidade de passageiros:");
        int N = scanner.nextInt();

        for (int i = 0; i < N; i++) {
            System.out.println("Você possui RG?");
            String possuiRG = scanner.next();
            System.out.println("Qual sua data de nascimento no RG?");
            String dataNascimentoRG = scanner.next();
            System.out.println("Você possui passagem?");
            String possuiPassagem = scanner.next();
            System.out.println("E a data de nascimento na passagem?");
            String dataNascimentoPassagem = scanner.next();
            System.out.println("Qual o seu assento?");
            String assento = scanner.next();

            if (possuiRG.equals("Nao")) {
                System.out.println("a saída é nessa direção");
            } else if (possuiPassagem.equals("Nao")) {
                System.out.println("a recepção é nessa direção");
            } else if (!dataNascimentoRG.equals(dataNascimentoPassagem)) {
                System.out.println("Infelizmente, terei que chamar a polícia.");
            } else {
                System.out.println("o seu assento é " + assento + " tenha um ótimo dia");
            }
        }
    }
}

File: Q01.hs
import Text.Printf

main :: IO ()
main = do
    putStrLn "Quantidade de passageiros: "
    n <- readLn :: IO Int
    processPassengers n

processPassengers :: Int -> IO ()
processPassengers 0 = return ()
processPassengers n = do
    putStrLn "Você possui RG? (Sim ou Nao): "
    possuiRG <- getLine
    putStrLn "Qual sua data de nascimento no RG? (DD/MM/AAAA): "
    dataNascimentoRG <- getLine
    putStrLn "Você possui passagem? (Sim ou Nao): "
    possuiPassagem <- getLine
    putStrLn "E a data de nascimento na passagem? (DD/MM/AAAA): "
    dataNascimentoPassagem <- getLine
    putStrLn "Qual o seu assento? "
    assento <- getLine
    if possuiRG == "Nao"
        then printf "A saída é nessa direção\n"
        else if possuiPassagem == "Nao"
            then printf "A recepção é nessa direção\n"
            else if dataNascimentoRG /= dataNascimentoPassagem
                then printf "Infelizmente terei que chamar a polícia.\n"
                else printf "O seu assento é %s, tenha um ótimo dia\n" assento
    processPassengers (n - 1)
